pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -244,12 +244,19 @@ func endpointHealthStatus(svc *model.Service, e v1.Endpoint) model.HealthStatus
 		return model.Healthy
 	}
 
+	// An endpoint is draining only if it was previously ready (serving == true) and persistent sessions is enabled
 	if svc != nil && svc.SupportsDrainingEndpoints() &&
 		(e.Conditions.Serving == nil || *e.Conditions.Serving) &&
 		(e.Conditions.Terminating == nil || *e.Conditions.Terminating) {
 		return model.Draining
 	}
 
+	// If it is shutting down, mark it as terminating. This occurs regardless of whether it was previously healthy or not.
```

Comment:
What's the observable behavior if all pods in the svc are marked as terminating?

Is the main visible change here that failover simply wouldn't happen as "early" (potentially) as it did previously?

Reply:
If the entire service is terminating, both before and after no endpoints are routable so you would get 503 UH.

If just one zone is entirely terminating:
Before: we are seen as 100% unhealthy in the local zone so failover
After: we have no endpoints at all in the local zone so we send to other zones